interface/reports/inventory_list.php

```diff
@@ -26,46 +26,662 @@
+        }
+        echo "  <td align='right'>" . text($row['on_hand'])               . "</td>\n";
+        echo "  <td align='right'>" . text($zerodays)                     . "</td>\n";
+        // Do not text() these next 2 lines.
+        echo "  <td align='right'>" . $monthly                            . "</td>\n";
+        echo "  <td align='right'>" . $stock_months                       . "</td>\n";
+        echo "  <td align='right'>" . text($reorder_qty)                  . "</td>\n";
+        // Do not text() the following line.
+        echo "  <td style='color:red'>" . $warnings                       . "</td>\n";
+        echo " </tr>\n";
+    } // end not exporting
+
+    $wrl_last_drug_id = $drug_id;
+}
+
+if (!empty($_POST['form_days'])) {
+    $form_days = $_POST['form_days'] + 0;
+} else {
+    $form_days = sprintf('%d', (strtotime(date('Y-m-d')) - strtotime(date('Y-01-01'))) / (60 * 60 * 24) + 1);
+}
+
 // this is "" or "submit".
-$form_action = $_POST['form_action'];
+$form_action = $_POST['form_action'] ?? '';
 
 if (!empty($_POST['form_days'])) {
     $form_days = $_POST['form_days'] + 0;
 } else {
     $form_days = sprintf('%d', (strtotime(date('Y-m-d')) - strtotime(date('Y-01-01'))) / (60 * 60 * 24) + 1);
 }
 
-// get drugs
-$res = sqlStatement("SELECT d.*, SUM(di.on_hand) AS on_hand " .
-  "FROM drugs AS d " .
-  "LEFT JOIN drug_inventory AS di ON di.drug_id = d.drug_id " .
-  "AND di.on_hand != 0 AND di.destroy_date IS NULL " .
-  "WHERE d.active = 1 " .
-  "GROUP BY d.name, d.drug_id ORDER BY d.name, d.drug_id");
-?>
+// this is "" or "submit" or "export".
+$form_action = empty($_POST['form_action']) ? '' : $_POST['form_action'];
+
+$form_inactive = empty($_REQUEST['form_inactive']) ? 0 : 1;
+$form_details = empty($_REQUEST['form_details']) ? 0 : intval($_REQUEST['form_details']);
+$form_facility = 0 + empty($_REQUEST['form_facility']) ? 0 : $_REQUEST['form_facility'];
+$form_consumable = isset($_REQUEST['form_consumable']) ? intval($_REQUEST['form_consumable']) : 0;
+$form_orderby = $_REQUEST['form_orderby'] ?? 'name';
+$orderby = $ORDERHASH[$form_orderby];
+
+// Incoming form_warehouse, if not empty is in the form "warehouse/facility".
+// The facility part is an attribute used by JavaScript logic.
+$form_warehouse = $_REQUEST['form_warehouse'] ?? '';
+$tmp = explode('/', $form_warehouse);
+$form_warehouse = $tmp[0];
+
+$mmtype = empty($GLOBALS['gbl_min_max_months']) ? xl('Units') : xl('Months');
+
+// Compute WHERE condition for filtering on facility/warehouse.
+$fwcond = '';
+if ($form_facility) {
+    $fwcond .= " AND lo.option_value IS NOT NULL AND lo.option_value = '" . add_escape_custom($form_facility) . "'";
+}
+if ($form_warehouse) {
+    $fwcond .= " AND di.warehouse_id IS NOT NULL AND di.warehouse_id = '" . add_escape_custom($form_warehouse) . "'";
+}
+
+$uwcond = $is_user_restricted ?
+    ("AND di.warehouse_id IS NOT NULL AND di.warehouse_id IN (" . genUserWarehouses() . ")") : "";
```

Comment:
need to escape
(rec looking at all these queries and trying to support binding in them and ensure that no sql injections get through these queries (the problem with segmented queries like these is risk of sql injection getting through))

Reply:
I'm double checking to make sure the code uses add_escape_custom() where appropriate. Agree this requires being careful. Doesn't binding translate to the same thing anyway?